modules/stats/drawdown/per_coin.py

```diff
@@ -1,16 +1,21 @@
 import pandas as pd
 import numpy as np
 
-from modules.stats.drawdown.drawdown import get_max_drawdown_ratio
 from modules.stats.trade import Trade
 
 
-def get_max_seen_drawdown_per_coin(signal_dict, closed_pair_trades: [Trade], fee_percentage: float):
+def get_cum_profit_ratio_per_coin(signal_dict, closed_pair_trades: [Trade], fee_percentage: float,
+                                  realised_drawdown=False):
     trades_opened_closed_timestamps = map_trades_to_opened_closed_timestamps(closed_pair_trades)
 
     values = signal_dict.values()
     df = pd.DataFrame(values).set_index("time")
 
+    # Check if we want to calculate realised drawdown
+    if realised_drawdown:
+        trade_timestamps = get_trade_timestamps(closed_pair_trades)
+        df = pd.concat([df, trade_timestamps], axis=1, join="inner")
+
```

Comment:
I don't agree with mixing the function here - sure, it's dryer, but it's also less readable. From the name of the function it's not clear what you're calculating exactly, best to split it back up in two more methods imho

Reply:
Yes I partly agree. I think using the same code twice only for the sake of readability is bad practice. However, I do agree that the function has become less readable. If I change the code to something suggested here, would that clear up the function?

```suggestion
    # Only keep timestamps of closed trades (open & close) if only_realised_profits is True.
    if only_realised_profits:
        trade_timestamps = get_trade_timestamps(closed_pair_trades)
        df = pd.concat([df, trade_timestamps], axis=1, join="inner")
```